charts/backstage/templates/sonataflows.yaml

```
@@ -0,0 +1,122 @@
+spec:
+  monitoring:
+    enabled: {{ .Values.orchestrator.sonataflowPlatform.monitoring.enabled }}
+  build:
+    template:
+      resources:
+        requests:
+          memory: {{ .Values.orchestrator.sonataflowPlatform.resources.requests.memory }}
+          cpu: {{ .Values.orchestrator.sonataflowPlatform.resources.requests.cpu }}
+        limits:
+          memory: {{ .Values.orchestrator.sonataflowPlatform.resources.limits.memory }}
+          cpu: {{ .Values.orchestrator.sonataflowPlatform.resources.limits.cpu }}
+  {{- if (and (.Values.orchestrator.sonataflowPlatform.eventing.broker.name) (.Values.orchestrator.sonataflowPlatform.eventing.broker.namespace)) }}
+  eventing:
+    broker:
+      ref:
+        apiVersion: eventing.knative.dev/v1
+        kind: Broker
+        name: {{ .Values.orchestrator.sonataflowPlatform.eventing.broker.name }}
+        namespace: {{ .Values.orchestrator.sonataflowPlatform.eventing.broker.namespace }}
+  {{- end }}
+  services:
+    dataIndex:
+      enabled: true
+      persistence:
+        postgresql:
+          {{- if .Values.upstream.postgresql.enabled }}
+          secretRef:
+            name: {{ .Release.Name }}-postgresql-svcbind-postgres
+            userKey: username
+            passwordKey: password
+          serviceRef:
+            name: {{ .Release.Name }}-postgresql
+            namespace: {{ .Release.Namespace }}
+            databaseName: sonataflow
+          {{- else }}
+          secretRef:
+            name: {{ .Values.orchestrator.sonataflowPlatform.externalDBsecretRef }}
+            userKey: POSTGRES_USER
+            passwordKey: POSTGRES_PASSWORD
+          jdbcUrl: jdbc:postgresql://{{ .Values.upstream.backstage.appConfig.backend.database.connection.host}}:{{ .Values.upstream.backstage.appConfig.backend.database.connection.port}}/sonataflow?currentSchema=data-index-service
+          {{- end }}
+      {{- if .Values.orchestrator.sonataflowPlatform.dataIndexImage }}
+      podTemplate:
+        container:
+          image: {{ .Values.orchestrator.sonataflowPlatform.dataIndexImage }}
+      {{- end }}
+    jobService:
+      enabled: true
+      persistence:
+        postgresql:
+          {{- if .Values.upstream.postgresql.enabled }}
+          secretRef:
+            name: {{ .Release.Name }}-postgresql-svcbind-postgres
+            userKey: username
+            passwordKey: password
+          serviceRef:
+            name: {{ .Release.Name }}-postgresql
+            namespace: {{ .Release.Namespace }}
+            databaseName: sonataflow
+          {{- else }}
+          secretRef:
+            name: {{ .Values.orchestrator.sonataflowPlatform.externalDBsecretRef}}
+            userKey: POSTGRES_USER
+            passwordKey: POSTGRES_PASSWORD
+          jdbcUrl: jdbc:postgresql://{{ .Values.upstream.backstage.appConfig.backend.database.connection.host}}:{{ .Values.upstream.backstage.appConfig.backend.database.connection.port}}/sonataflow?currentSchema=jobs-service
```

```suggestion
          jdbcUrl: jdbc:postgresql://{{ .Values.upstream.backstage.appConfig.backend.database.connection.host}}:{{ .Values.upstream.backstage.appConfig.backend.database.connection.port}}/{{ .Values.orchestrator.sonataflowPlatform.externalDBName }}?currentSchema=jobs-service
```

Should be the configurable `externalDBName`, no?

---

Will the `jobs-service` schema be created automatically in the external database? Or is the responsibility of the admin to create it? If so, I guess it should be documented accordingly.

---

This is created automatically, yes